src/app/components/AdvancedSearch/AdvancedSearch.jsx

```diff
@@ -215,9 +208,21 @@ class AdvancedSearch extends React.Component {
     if (!searchQuery) {
       return;
     }
+    if (searchQuery.showField && searchQuery.showQuery) {
+      state.queries[searchQuery.showField] = searchQuery.showQuery;
+    }
+
     if (searchQuery.queries) {
       searchQuery.queries.forEach((q) => {
-        state.queries[q.field] = q.query;
+        const allowedFields = inputTerms.map(term => term.values.map(value => value.key)).flat();
+        if (allowedFields.indexOf(q.field) >= 0) {
+          // If field is already set (by showQuery), use the showQuery value.
```

Comment:
If the show query is an author name, I think this should still query with the VIAF identifier? I'm not 100% clear on what this is doing though, so maybe that's already addressed. Otherwise this is a safe assumption to make.

Reply:
You're right:  Right now if someone clicks the author name, it looks it up by viaf, but if they hit advanced search and then hit search again, it will only search by name.  

Is there a way to look up the author/publisher from the viaf?  I'm worried about a situation where someone clicks an author name, so it searches by viaf, then they change the name in advanced search, but the viaf (for the original author) still persists.  How do we check that the viaf still pertains to the author?

Reply:
hmm, good point, there is an endpoint to look up VIAF identifiers, but I'm not sure we should be using that from the front end...

could you do it from the `onQueryChange` method? Check if the field is author and then wipe the viaf and lcnaf fields in state?

Reply:
Oh that's a great idea!  Let's do that!